packages/editor/src/components/autosave-monitor/index.native.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * External dependencies
+ */
+import RNReactNativeGutenbergBridge from 'react-native-gutenberg-bridge';
+
+export class AutosaveMonitor extends Component {
+	componentDidUpdate( prevProps ) {
+		const { isDirty, editsReference, isAutosaving } = this.props;
+
+		// The edits reference is held for comparison to avoid scheduling an
+		// autosave if an edit has not been made since the last autosave
+		// completion. This is assigned when the autosave completes, and reset
+		// when an edit occurs.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/12318
+
+		if ( editsReference !== prevProps.editsReference ) {
+			this.didAutosaveForEditsReference = false;
+		}
+
+		if ( ! isAutosaving && prevProps.isAutosaving ) {
+			this.didAutosaveForEditsReference = true;
+		}
+
+		if (
+			prevProps.isDirty !== isDirty ||
+			prevProps.editsReference !== editsReference
+		) {
+			this.toggleTimer(
+				isDirty &&
+				! this.didAutosaveForEditsReference
+			);
+		}
+	}
+
+	componentWillUnmount() {
+		this.toggleTimer( false );
+	}
+
+	toggleTimer( isPendingSave ) {
+		clearTimeout( this.pendingSave );
+		const { autosaveInterval } = this.props;
+
+		if ( isPendingSave ) {
+			this.pendingSave = setTimeout(
+				() => {
+					RNReactNativeGutenbergBridge.editorDidAutosave();
```

Comment:
Yes, I needed to remove the references to `isEditedPostAutosaveable` since it uses the network stack.

https://github.com/WordPress/gutenberg/blob/9c82ea0967e016cdc12004c91c9601eba81bcf5f/packages/editor/src/components/autosave-monitor/index.js#L34

https://github.com/WordPress/gutenberg/blob/9c82ea0967e016cdc12004c91c9601eba81bcf5f/packages/editor/src/components/autosave-monitor/index.js#L73

Reply:
Just an idea:
We could `import { AutosaveMonitor } from './index.js'`, it would import just the component without the HOC composition, then we add our native HOC composition and the extra props from the parent.